joomla/joomla-cms
# [4.0] Implementation of disable in joomla-field-switcher

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/20852.

### Summary of Changes

- Add  attributes to `joomla-field-switcher` if field is set to `disable="true"`

- Make the field `joomla-field-switcher` if field is set to `disable="true"`



### Testing Instructions

Open global configuration and see outer html of the field "Search Engine Friendly URLs"
```

<joomla-field-switcher id="jform_sef" off-text="No" on-text="Yes" type="success">
<span class="switcher active has-success" tabindex="0" aria-checked="true" role="switch" aria-label="Yes"><input id="jform_sef0" name="jform[sef]" value="0" tabindex="-1" class="valid form-control-success" aria-invalid="false" type="radio"><input id="jform_sef1" name="jform[sef]" value="1" class="valid active form-control-success" tabindex="-1" checked="" aria-invalid="false" type="radio"><span class="switch"></span></span><span class="switcher-labels"><span class="switcher-label-0">No</span><span class="switcher-label-1 active">Yes</span></span></joomla-field-switcher>
```

Open `/administrator/components/com_config/forms/application.xml`

and change the field (insert `disabled="true"`)

```
		<field
			name="sef"
			type="radio"
			label="COM_CONFIG_FIELD_SEF_URL_LABEL"
			class="switcher"
			default="1"
			filter="boolean"
			>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>

```
to 

```
		<field
			name="sef"
			type="radio"
			label="COM_CONFIG_FIELD_SEF_URL_LABEL"
			class="switcher"
			disabled="true"
			default="1"
			filter="boolean"
			>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>
```

Open again global configuration and see outer html of the field "Search Engine Friendly URLs". Now the text `disabled=""`  is added.


Apply this patch and see, that the text `disabled="disabled"`  is added to the `joomla-field-switcher` and the `inputs` of  `joomla-field-switcher`. See also, that the field now works as expected.



See also Issue https://github.com/joomla/joomla-cms/issues/20852.


### Expected result
The field "Search Engine Friendly URLs" is disabled. You can not change it.


### Actual result
Nothing changed after making the field disabled. The field works as before. You can change the value. 





### Documentation Changes Required
Yes.
